pyomo/solvers/tests/checks/test_CPLEXDirect.py

```diff
@@ -143,5 +143,88 @@ def test_optimal_mip(self):
             self.assertEqual(results.solution.status,
                              SolutionStatus.optimal)
 
+
+@unittest.skipIf(not unittest.mock_available, "'mock' is not available")
+@unittest.skipIf(not cplexpy_available, "The 'cplex' python bindings are not available")
+class TestIsFixedCallCount(unittest.TestCase):
+    def setup(self, skip_trivial_constraints):
+        m = ConcreteModel()
+        m.x = Var()
+        m.y = Var()
+        m.c1 = Constraint(expr=m.x + m.y == 1)
+        m.c2 = Constraint(expr=m.x <= 1)
+        self.assertFalse(m.c2.has_lb())
+        self.assertTrue(m.c2.has_ub())
+        self._model = m
+
+        self._opt = SolverFactory("cplex_persistent")
+        self._opt.set_instance(
+            self._model, skip_trivial_constraints=skip_trivial_constraints
+        )
+
+    def test_skip_trivial_and_call_count_for_fixed_con_is_one(self):
+        self.setup(skip_trivial_constraints=True)
+        self._model.x.fix(1)
+        self.assertTrue(self._opt._skip_trivial_constraints)
+        self.assertTrue(self._model.c2.body.is_fixed())
+
+        with unittest.mock.patch(
+            "pyomo.solvers.plugins.solvers.cplex_direct.is_fixed", wraps=is_fixed
+        ) as mock_is_fixed:
+            mock_is_fixed.assert_not_called()
+            self._opt.add_constraint(self._model.c2)
+            mock_is_fixed.assert_called_once()
```

Comment:
Note that `assert_not_called` was added in Python 3.5 and `assert_called_once` was added in Python 3.6, so we are seeing test failures on Python 3.4 and 3.5.

Reply:
Good point. I've updated to rely solely on `call_count`